modin/pandas/dataframe.py

```diff
@@ -819,12 +819,57 @@ def explode(
             pandas.DataFrame.explode, column, ignore_index=ignore_index
         )
 
+    def _update_var_dicts_in_kwargs(self, expr, **kwargs):
+        """
+        Copy variables with "@" prefix in `local_dict` and `global_dict`.
+
+        Parameters
+        ----------
+        expr : str
+            The expression string to search variables with "@" prefix.
+        **kwargs : dict
+            See the documentation for eval() for complete details on the keyword arguments accepted by query().
+
+        Returns
+        -------
+        kwargs : dict
+            Dict with updated `local_dict` and `global_dict` keys.
+        """
+        import re
+
+        frame = sys._getframe()
+        try:
+            f_locals = frame.f_back.f_back.f_locals
+            f_globals = frame.f_back.f_back.f_globals
+        finally:
+            del frame
+        local_names = re.findall(r"@([\w]*)", expr)
+        local_dict = {}
+        global_dict = {}
+
+        for name in local_names:
+            if name in f_locals:
+                local_dict[name] = f_locals[name]
+            elif name in f_globals:
+                global_dict[name] = f_globals[name]
+
+        if local_dict:
+            if kwargs.get("local_dict"):
+                local_dict.update(kwargs["local_dict"])
+            kwargs["local_dict"] = local_dict
+        if global_dict:
+            if kwargs.get("global_dict"):
+                global_dict.update(kwargs.get("global_dict"))
```

Comment:
Squash the ifs like this:
```suggestion
        if local_dict and kwargs.get("local_dict"):
            local_dict.update(kwargs["local_dict"])
            kwargs["local_dict"] = local_dict
        if global_dict and kwargs.get("global_dict"):
            global_dict.update(kwargs["global_dict"])
```

Reply:
In case when `kwargs.get("local_dict")==None` and `local_dict=={"local_var": 10}` we should make the assignment: `kwargs["local_dict"] = local_dict` but with these changes it won't.

Reply:
Right... in this case I suggest to change it to
```python
        if local_dict:
            local_dict.update(kwargs.get("local_dict") or {})
            kwargs["local_dict"] = local_dict
        if global_dict:
            global_dict.update(kwargs.get("global_dict") or {})
```
(note the `or {}` part - this trick pushes the creation of an empty dictionary to the moment in time when a key is missing in `kwargs` - a lazier version of `.get(key, {})`)